lib/IDE/IDETypeChecking.cpp

```diff
@@ -269,71 +271,90 @@ struct SynthesizedExtensionAnalyzer::Implementation {
   }
 
   std::pair<SynthesizedExtensionInfo, ExtensionMergeInfo>
-  isApplicable(ExtensionDecl *Ext, bool IsSynthesized) {
-    SynthesizedExtensionInfo Result(IsSynthesized);
+  isApplicable(ExtensionDecl *Ext, bool IsSynthesized,
+               ExtensionDecl *EnablingExt, NormalProtocolConformance *Conf) {
+    SynthesizedExtensionInfo Result(IsSynthesized, EnablingExt);
     ExtensionMergeInfo MergeInfo;
     MergeInfo.HasDocComment = !Ext->getRawComment().isEmpty();
     MergeInfo.InheritsCount = countInherits(Ext);
-    if (!Ext->isConstrainedExtension()) {
+    auto conformanceIsConditional =
+        Conf && !Conf->getConditionalRequirements().empty();
+    if (!Ext->isConstrainedExtension() && !conformanceIsConditional) {
```

Comment:
It seems to me that one cannot have a conditional conformance without also having a constrained extension?

Reply:
It's confusing (took me 10 minutes to remember why I had to have this... will add a comment), but they're referring to two different extensions:

```swift
protocol Foo {}
extension Foo { ... } // Ext

struct Bar<T> {}
extension Bar : Foo where T: Foo {} // Conf
```

Reply:
Oh, of course! `Ext` is the protocol extension we're considering, and `Conf` is referring to a nominal type's conditional conformance to the corresponding protocol.